File: Assignment37_1.cpp
//1.Write a program in C++ which checks whether 15th bit is On or OFF. 

#include<iostream>
using namespace std;

class BitOpration
{
    public:
        int iNo;
        int iMask;
        int iResult;

        BitOpration(int A)
        {
            iNo = A;
            iMask = 0X00004000;
            iResult = 0;
        }

        bool ChkBit()
        {
            iResult = iNo & iMask;

            if(iMask == iResult)
            {
                return true;
            }
            else
            {
                return false;
            }

        }

};
int main()
{
    bool bRet = false;
    int iValue = 0;

    cout<<"Enter the Number Here : \n";
    cin>>iValue;
    
    BitOpration obj(iValue);

    bRet = obj.ChkBit();

    if(bRet == true)
    {
        cout<<"15th Bit is ON....\n";
    }
    else
    {
        cout<<"15th Bit is OFF....\n";
    }

    return 0;
}

File: Assignment37_1.py
#1.Write a program in C++ which checks whether 15th bit is On or OFF. 

class BitOpr:
    
    iValue = None
    iMask = None
    iResult = None
    
    def __init__(self,iValue):
        self.iValue = iValue
        self.iMask = 0X00004000
        self.iResult = 0
    
    def ChkBit(self):
        
        self.iResult = self.iValue & self.iMask
        
        if(self.iResult == self.iMask):
            return True
        else:
            return False
            
                  

def main():
    
    iNo = 0
    bRet = None
    
    print("Enter the Number Here : ")
    iNo = int(input())
    
    obj = BitOpr(iNo)
    bRet = obj.ChkBit()
    
    if(bRet == True):
        print("The 15th Bit is ON...")
    else:
        print("15th Bit is OFF...")
        
       


if __name__=="__main__":
    main()